stl/inc/ranges

```diff
@@ -972,7 +964,7 @@ namespace ranges {
 
             template <_Integer_like _Ty>
             _NODISCARD constexpr auto operator()(_Ty _Length) const noexcept {
-                return _Partial<_Ty>{_Length};
+                return _Partial<_Ty>{._Length = _Length};

```

Comment:
Is this the first use of designated initializers in the STL? Is this necessary? I don't see any other data members. Occurs below.

(I am slightly nervous about compiler bugs, but I don't object to shipping this.)

Reply:
Yes, I believe this is the first use. Designated initializers here enable us to avoid explicitly initializing the base class a la `_Partial<_Ty>{{}, _Length}`, which is nice but not critical. The fact that we don't really _need_ the feature here and that the change to avoid using it is so simple suggests to me that this is a good place to pilot using that feature in the STL: if there's a problem we're minutes away from a fix rather than hours or days.